### PR TITLE
feat(mechanism): optimize address parsing of stake interfaces

### LINKED ISSUE #5411: Optimize address parsing of stake-related interfaces
# Rationale
TransactionLog combined with EventPlugin mechanism is an important way to analyze transaction data.

At present, it is found that when calling the query transaction interface of the tron-eventquery service,
one of the transaction's fields returned is as follows:  
```
{"contractType": "UnDelegateResourceContract","fromAddress": null,"toAddress": null}
``` 

To the transaction type: UnDelegateResourceContract, the returned fields: fromAddress, toAddress are not correctly assigned.

Below are the  stake-related transactions include:
- UnfreezeBalanceContract
- FreezeBalanceV2Contract
- UnfreezeBalanceV2Contract
- WithdrawExpireUnfreezeContract
- DelegateResourceContract
- UnDelegateResourceContract
- CancelAllUnfreezeV2Contract

The fromAddress and toAddress should be correctly parsed into the TransactionLog, and then the fields will be correct when the query transaction of the tron-eventquery service is called.

# Why should this feature exist?

Before writing the new transaction into the transactionLog, the address needs to be parsed correctly
    
# Implementation    

For the new transactions, to correctly parse fromAddress and toAddress to TransactionLog, the following code must be modified: class TransactionLogTriggerCapsule, method TransactionLogTriggerCapsule

```java
public TransactionLogTriggerCapsule(TransactionCapsule trxCapsule, BlockCapsule blockCapsule,
       int txIndex, long preCumulativeEnergyUsed, long preCumulativeLogCount,
       TransactionInfo transactionInfo, long energyUnitPrice) {
           case UnfreezeBalanceContract:
               ...
           break;
           case FreezeBalanceV2Contract:
                …
           break;
           case UnfreezeBalanceV2Contract:
                …
           break;
           case WithdrawExpireUnfreezeContract:
                …
           break;
           case DelegateResourceContract:
                …
           break;
           case UnDelegateResourceContract:
                …
           break;
           case CancelAllUnfreezeV2Contract:
                …
           break;
}
``` 

     
In this way, for the newly added transactions, fromAddress and toAddress are able to be correctly parsed and then assigned to TransactionLogTrigger.

### PR DESCRIPTION
**Pull request by CarlChaoCarl**
**What does this PR do?**
optimize address parsing of stake  interfaces, close #5411

**Why are these changes required?**

**This PR has been tested by:**
- Unit Tests
- Manual Testing

**Follow up**

**Extra details**

### CODE REVIEW

File: framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java [hunk 1]
```diff
@@ -156,6 +163,100 @@ public TransactionLogTriggerCapsule(TransactionCapsule trxCapsule, BlockCapsule
...
+            case UnfreezeBalanceContract:
+              UnfreezeBalanceContract unfreezeBalanceContract = contractParameter
+                  .unpack(UnfreezeBalanceContract.class);
+
+              if (Objects.nonNull(unfreezeBalanceContract)) {
+                if (Objects.nonNull(unfreezeBalanceContract.getOwnerAddress())) {
+                  transactionLogTrigger.setFromAddress(StringUtil
+                      .encode58Check(unfreezeBalanceContract.getOwnerAddress().toByteArray()));
+                }
+                if (Objects.nonNull(unfreezeBalanceContract.getReceiverAddress())) {
```
Comment (wubin01) [inline framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:175]: The frequency of these if statements is too high, you can encapsulate a function.

File: framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java [hunk 2]
```diff
@@ -156,6 +163,100 @@ public TransactionLogTriggerCapsule(TransactionCapsule trxCapsule, BlockCapsule
...
+              break;
+            case WithdrawExpireUnfreezeContract:
+              WithdrawExpireUnfreezeContract withdrawExpireUnfreezeContract = contractParameter
+                  .unpack(WithdrawExpireUnfreezeContract.class);
+
+              if (Objects.nonNull(withdrawExpireUnfreezeContract)
+                  && Objects.nonNull(withdrawExpireUnfreezeContract.getOwnerAddress())) {
+                transactionLogTrigger.setFromAddress(StringUtil.encode58Check(
+                      withdrawExpireUnfreezeContract.getOwnerAddress().toByteArray()));
+              }
```
Comment (tomatoishealthy) [inline framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:214]: Better to add the affected balance amount? same with other contracts which also not set the balance field.

File: framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java [hunk 3]
```diff
@@ -156,6 +163,100 @@ public TransactionLogTriggerCapsule(TransactionCapsule trxCapsule, BlockCapsule
...
+            case FreezeBalanceV2Contract:
+              FreezeBalanceV2Contract freezeBalanceV2Contract = contractParameter
+                  .unpack(FreezeBalanceV2Contract.class);
+
+              if (Objects.nonNull(freezeBalanceV2Contract)) {
+                if (Objects.nonNull(freezeBalanceV2Contract.getOwnerAddress())) {
+                  transactionLogTrigger.setFromAddress(StringUtil
+                      .encode58Check(freezeBalanceV2Contract.getOwnerAddress().toByteArray()));
+                }
+                transactionLogTrigger.setAssetAmount(freezeBalanceV2Contract.getFrozenBalance());
```
Comment (halibobo1205) [inline framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:190]: `transactionLogTrigger.setAssetName("trx");` ?

File: framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java [hunk 4]
```diff
@@ -156,6 +163,100 @@ public TransactionLogTriggerCapsule(TransactionCapsule trxCapsule, BlockCapsule
...
+              if (Objects.nonNull(delegateResourceContract)) {
+                if (Objects.nonNull(delegateResourceContract.getOwnerAddress())) {
+                  transactionLogTrigger.setFromAddress(StringUtil
+                      .encode58Check(delegateResourceContract.getOwnerAddress().toByteArray()));
+                }
+                if (Objects.nonNull(delegateResourceContract.getReceiverAddress())) {
+                  transactionLogTrigger.setToAddress(StringUtil
+                      .encode58Check(delegateResourceContract.getReceiverAddress().toByteArray()));
+                }
+                transactionLogTrigger.setAssetAmount(
```
Comment (halibobo1205) [inline framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:229]: `transactionLogTrigger.setAssetName("trx");`?

Comment (CarlChaoCarl) [reply framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:229]: OK，I get it.

File: framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java (hunk 2 above)
Comment (CarlChaoCarl) [reply framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:214]: OK，I add it.

File: framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java (hunk 1 above)
Comment (CarlChaoCarl) [reply framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:175]: OK, Ok, I encapsulate the code

File: framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java [hunk 5]
```diff
@@ -156,6 +163,90 @@ public TransactionLogTriggerCapsule(TransactionCapsule trxCapsule, BlockCapsule
...
+              }
+              break;
+            case WithdrawExpireUnfreezeContract:
+              WithdrawExpireUnfreezeContract withdrawExpireUnfreezeContract = contractParameter
+                  .unpack(WithdrawExpireUnfreezeContract.class);
+
+              if (Objects.nonNull(withdrawExpireUnfreezeContract)) {
+                transactionLogTrigger.setFromAddress(StringUtil.encode58Check(
+                      withdrawExpireUnfreezeContract.getOwnerAddress().toByteArray()));
+                transactionLogTrigger.setAssetAmount(transactionInfo.getWithdrawExpireAmount());
```
Comment (halibobo1205) [inline framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:209]: transactionLogTrigger.setAssetName("trx");

File: framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java [hunk 6]
```diff
@@ -156,6 +163,90 @@ public TransactionLogTriggerCapsule(TransactionCapsule trxCapsule, BlockCapsule
...
+                transactionLogTrigger.setAssetName("trx");
+                transactionLogTrigger.setAssetAmount(
+                      unfreezeBalanceV2Contract.getUnfreezeBalance());
+              }
+              break;
+            case WithdrawExpireUnfreezeContract:
+              WithdrawExpireUnfreezeContract withdrawExpireUnfreezeContract = contractParameter
+                  .unpack(WithdrawExpireUnfreezeContract.class);
+
+              if (Objects.nonNull(withdrawExpireUnfreezeContract)) {
```
Comment (halibobo1205) [inline framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:206]: contract cannot be null.

File: framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java [hunk 7]
```diff
@@ -156,6 +163,90 @@ public TransactionLogTriggerCapsule(TransactionCapsule trxCapsule, BlockCapsule
                     StringUtil.encode58Check(createSmartContract.getOwnerAddress().toByteArray()));
               }
               break;
+            case UnfreezeBalanceContract:
+              UnfreezeBalanceContract unfreezeBalanceContract = contractParameter
+                  .unpack(UnfreezeBalanceContract.class);
+
+              if (Objects.nonNull(unfreezeBalanceContract)) {
```
Comment (halibobo1205) [inline framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:170]: contract cannot be null.

File: framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java [hunk 8]
```diff
@@ -156,6 +163,90 @@ public TransactionLogTriggerCapsule(TransactionCapsule trxCapsule, BlockCapsule
...
+                  transactionLogTrigger.setToAddress(StringUtil
+                        .encode58Check(unfreezeBalanceContract.getReceiverAddress().toByteArray()));
+                }
+              }
+              break;
+            case FreezeBalanceV2Contract:
+              FreezeBalanceV2Contract freezeBalanceV2Contract = contractParameter
+                  .unpack(FreezeBalanceV2Contract.class);
+
+              if (Objects.nonNull(freezeBalanceV2Contract)) {
```
Comment (halibobo1205) [inline framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:183]: contract cannot be null.

File: framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java [hunk 9]
```diff
@@ -156,6 +163,90 @@ public TransactionLogTriggerCapsule(TransactionCapsule trxCapsule, BlockCapsule
...
+                      .encode58Check(freezeBalanceV2Contract.getOwnerAddress().toByteArray()));
+                transactionLogTrigger.setAssetName("trx");
+                transactionLogTrigger.setAssetAmount(freezeBalanceV2Contract.getFrozenBalance());
+              }
+              break;
+            case UnfreezeBalanceV2Contract:
+              UnfreezeBalanceV2Contract unfreezeBalanceV2Contract = contractParameter
+                  .unpack(UnfreezeBalanceV2Contract.class);
+
+              if (Objects.nonNull(unfreezeBalanceV2Contract)) {
```
Comment (halibobo1205) [inline framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:194]: contract cannot be null.

File: framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java [hunk 10]
```diff
@@ -156,6 +163,90 @@ public TransactionLogTriggerCapsule(TransactionCapsule trxCapsule, BlockCapsule
...
+                transactionLogTrigger.setFromAddress(StringUtil.encode58Check(
+                      withdrawExpireUnfreezeContract.getOwnerAddress().toByteArray()));
+                transactionLogTrigger.setAssetAmount(transactionInfo.getWithdrawExpireAmount());
+              }
+              break;
+            case DelegateResourceContract:
+              DelegateResourceContract delegateResourceContract = contractParameter
+                  .unpack(DelegateResourceContract.class);
+
+              if (Objects.nonNull(delegateResourceContract)) {
```
Comment (halibobo1205) [inline framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:216]: contract cannot be null.

File: framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java [hunk 11]
```diff
@@ -156,6 +163,90 @@ public TransactionLogTriggerCapsule(TransactionCapsule trxCapsule, BlockCapsule
...
+                transactionLogTrigger.setAssetName("trx");
+                transactionLogTrigger.setAssetAmount(
+                    unDelegateResourceContract.getBalance());
+              }
+              break;
+            case CancelAllUnfreezeV2Contract:
+              CancelAllUnfreezeV2Contract cancelAllUnfreezeV2Contract = contractParameter
+                  .unpack(CancelAllUnfreezeV2Contract.class);
+
+              if (Objects.nonNull(cancelAllUnfreezeV2Contract)) {
```
Comment (halibobo1205) [inline framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:245]: contract cannot be null.

Comment (CarlChaoCarl) [reply framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:245]: OK, I get it

File: framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java [hunk 12]
```diff
@@ -156,6 +163,90 @@ public TransactionLogTriggerCapsule(TransactionCapsule trxCapsule, BlockCapsule
...
+                transactionLogTrigger.setAssetName("trx");
+                transactionLogTrigger.setAssetAmount(
+                      delegateResourceContract.getBalance());
+              }
+              break;
+            case UnDelegateResourceContract:
+              UnDelegateResourceContract unDelegateResourceContract = contractParameter
+                  .unpack(UnDelegateResourceContract.class);
+
+              if (Objects.nonNull(unDelegateResourceContract)) {
```
[the remark this answers is not in the document]
Comment (CarlChaoCarl) [reply framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:230]: OK, I get it

File: framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java (hunk 10 above)
Comment (CarlChaoCarl) [reply framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:216]: OK, I get it

File: framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java (hunk 9 above)
Comment (CarlChaoCarl) [reply framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:194]: OK, I get it

File: framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java (hunk 8 above)
Comment (CarlChaoCarl) [reply framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:183]: OK, I get it

File: framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java (hunk 7 above)
Comment (CarlChaoCarl) [reply framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:170]: OK, I get it

File: framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java (hunk 5 above)
Comment (CarlChaoCarl) [reply framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:209]: OK, I get it

File: framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java [hunk 13]
```diff
@@ -156,6 +163,77 @@ public TransactionLogTriggerCapsule(TransactionCapsule trxCapsule, BlockCapsule
...
+            case UnfreezeBalanceContract:
+              UnfreezeBalanceContract unfreezeBalanceContract = contractParameter
+                  .unpack(UnfreezeBalanceContract.class);
+
+              transactionLogTrigger.setFromAddress(StringUtil
+                  .encode58Check(unfreezeBalanceContract.getOwnerAddress().toByteArray()));
+              if (!ByteString.EMPTY.equals(unfreezeBalanceContract.getReceiverAddress())) {
+                transactionLogTrigger.setToAddress(StringUtil
+                    .encode58Check(unfreezeBalanceContract.getReceiverAddress().toByteArray()));
+              }
```
Comment (halibobo1205) [inline framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:175]: Does this need to be added? 
```java
 transactionLogTrigger.setAssetName("trx");
           transactionLogTrigger.setAssetAmount(transactionInfo.getUnfreezeAmount());
```

Comment (CarlChaoCarl) [reply framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java:175]: OK, I get it

**Review by ss3344520 — APPROVED** [review]

**Pull request merged**